src/app.ts

```diff
@@ -216,15 +218,20 @@ class App {
         const updatedTokenSet: TokenSet = await xero.disconnect(req.session.activeTenant.id)
         await xero.updateTenants()
 
-        const decodedIdToken: XeroIdToken = jwtDecode(updatedTokenSet.id_token);
-        const decodedAccessToken: XeroAccessToken = jwtDecode(updatedTokenSet.access_token)
-
-        req.session.decodedIdToken = decodedIdToken
-        req.session.decodedAccessToken = decodedAccessToken
-        req.session.tokenSet = updatedTokenSet;
-        req.session.allTenants = xero.tenants
-        req.session.activeTenant = xero.tenants[0]
-
+        if (xero.tenants.length > 0){
+          const decodedIdToken: XeroIdToken = jwtDecode(updatedTokenSet.id_token);
+          const decodedAccessToken: XeroAccessToken = jwtDecode(updatedTokenSet.access_token)
+          req.session.decodedIdToken = decodedIdToken
+          req.session.decodedAccessToken = decodedAccessToken
+          req.session.tokenSet = updatedTokenSet;
+          req.session.allTenants = xero.tenants
+          req.session.activeTenant = xero.tenants[0]
+        } else {
+          req.session.decodedIdToken= undefined
+          req.session.decodedAccessToken = undefined
+          req.session.allTenants = undefined
+          req.session.activeTenant = undefined
+        }
```

Comment:
This fixes the UI in case you "Disconnect" and its the last tenant connected. Otherwise if you disconnect your active tenant switches to the first in the `tenant` array